django_video_encoder/models.py

```diff
@@ -44,11 +65,17 @@ class Format(models.Model):
 
     extra_info = models.TextField("Encoder information (JSON)", blank=True)
 
+    def save(self, **kwargs):
+        if not self.format_label:
+            self.format_label = codec_width_height_to_format_label(
+                self.video_codec, self.width, self.height
+            )
+        super().save(**kwargs)
+
```

Comment:
This auto-define brings some difficulties with it, maybe it was not such a great idea and could be let as it is/was. I thought it could help when setting the formats...

Maybe I should avoid this change. However, it should be checked for uniqueness, which is not so easy as checks aren't ran... :-/

Reply:
What exactly are the difficulties this change causes? 

Just to be sure I get this right: formats are only created by the entries one makes into DJANGO_VIDEO_ENCODER_FORMATS list in settings.py?

If the checks don't work, couldn't you then just add a combined unique constraint (on video_codec, width and height) on the database / model  level.

But more likely than not I'm completely missing the point here ...

Reply:
It would be a super unique_together (video_codec, width and height, content_type_id, object_id) but it wouldn't solve the problem...

Because these values are filled by getting the video from the api, after a job has been sent, creating formats directly from the dict, not using any database constraints. This is why checks would really help

Reply:
I've changed the behaviour in a more recent version to make the behaviour more predictible